composer/trainer/trainer.py

```diff
@@ -908,18 +979,47 @@ def __init__(
                 # that need to be moved onto GPU.
             optimizers = map_collection(optimizers, self._device.optimizer_to_device)
 
-        # Grad Accum
-        auto_grad_accum = _is_auto_grad_accum(grad_accum, device=self._device)
-        if auto_grad_accum and profiler:
-            raise ValueError("`grad_accum='auto'` is not compatible with the profiler. It is recommended to run "
-                             "a mini-run with `grad_accum='auto'` to identify the optimal grad_accum value and "
-                             'then manually specify that in a second run with profiler.')
-        grad_accum = _get_initial_grad_accum(grad_accum)
+        # Microbatching
+        # To support backwards compatability, we currently support both train_device_microbatch_size
+        # and grad_accum. If both are specified, we will use train_device_microbatch_size.
+        if train_device_microbatch_size:
+            using_device_microbatch_size = True
+            if grad_accum != 1:
+                raise ValueError(
```

Comment:
"If both are specified, we will use train_device_microbatch_size." , but here if both are specified you error out

Reply:
Updated to be more clear. If grad_accum=1, we ignore it if train_device_microbatch_size is specified